pyannote/audio/pipelines/speech_separation.py

```diff
@@ -698,6 +698,15 @@ def apply(
         # strings and integers when reference is available and some hypothesis
         # speakers are not present in the reference)
 
+        # re-order sources so that they match
+        # the order given by diarization.labels()
+        inverse_mapping = {label: index for index, label in mapping.items()}
+        original_sliding_window = sources.sliding_window
+        data = sources.data[
+            :, [inverse_mapping[label] for label in diarization.labels()]
+        ]
+        sources = SlidingWindowFeature(data, original_sliding_window)
+
```

Comment:
```suggestion
        # re-order sources so that they match
        # the order given by diarization.labels()
        inverse_mapping = {label: index for index, label in mapping.items()}
        source.data = sources.data[
            :, [inverse_mapping[label] for label in diarization.labels()]
        ]

```